core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatEntry.scala

```diff
@@ -399,7 +398,7 @@ final class ColumnFormatValue
   override def toData(out: DataOutput): Unit = {
     val buffer = getBufferRetain
     try {
-      val numBytes = buffer.remaining()
+      val numBytes = buffer.limit()
```

Comment:
Some basic question. What happens when a thread tries to read the buffer when it is getting serialized, especially during overflow.